docroot/modules/humsci/hs_dashboard/src/AnnouncementsManager.php

```diff
@@ -0,0 +1,291 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\hs_dashboard;
+
+use Drupal\Core\Datetime\DateFormatterInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\File\FileExists;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\Core\Url;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class to handle HDSP Announcements.
+ */
+class AnnouncementsManager implements ContainerInjectionInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Announcement CSV location.
+   */
+  const ANNOUNCEMENTS_CSV = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vTQzSuPudq048D1NadRBE9h_s_-w-o4YtcC6AHfCdcqn3gX52akZNOaF5KAG9SeXkCV6PvIVmRtQ0HR/pub?gid=1146337887&single=true&output=csv';
```

Comment:
Could this be configurable instead of hard-coded? Doesn't have to be exposed via a form but then it could be changed with a config set and not a code push if ever necessary.